[Refactor] Specify gradle plugins declaratively in example.

panmari opened the pull request:
As opposed to declaratively. This is necessary to run with new java/gradle versions. See https://docs.flutter.dev/release/breaking-changes/flutter-gradle-plugin-apply. Jointly bumping version numbers for gradle and related for easier in a new flutter installation.

Tested by successfully compiling/installing example app on Android.

chipweinberger commented:
you can just delete all the files and run this

```
cd ./example                                                                  
flutter config --enable-android 
flutter create .
flutter run
```

this way we are up to date with the latest recommendation

panmari commented:
Done, the biggest difference was the additional move to kotlin.

chipweinberger commented:
you have extra linux changes here

chipweinberger commented:
Also, perhaps we should just delete all these files and change the docs to look like this:

<img width="264" alt="Screenshot 2025-02-22 at 12 54 10 PM" src="https://github.com/user-attachments/assets/15ad0c37-5cb9-4f26-8361-e05d759f9915" />

ekuleshov commented:
> Also, perhaps we should just delete all these files and change the docs ...

@chipweinberger but what about Android manifest settings for bluetooth in the example app?

panmari commented:
Rebased and removed linux changes. I feel it's easier to keep the boilerplate code to allow easily running the example. Outside of big moves, like the gradle declaritive style I point out in the description, updates won't be needed.

chipweinberger commented:
> > Also, perhaps we should just delete all these files and change the docs ...
> 
> @chipweinberger but what about Android manifest settings for bluetooth in the example app?

good point. maybe we could just leave those files.

panmari commented:
Either solution seems fine and an improvement to the status quo. Do you want me to change the PR to doc-approached instead?

One minor consideration: If we expect folks to regenerate files that are not checked in to version control, the working directory will always be slightly polluted for contributors.

chipweinberger commented:
> maybe we could just leave those files.

we just need to test if this will work.

> the working directory will always be slightly polluted for contributors.

we can use gitignore

chipweinberger commented:
So I tried the doc, approach  and it works. 

But we also need to change minSdkVersion from `minSdkVersion flutter.minSdkVersion` to `minSdkVersion 21`, so it requires 2 changes.
